components/SerialServer/src/serial.c

```diff
@@ -21,90 +21,48 @@
 #include <platsupport/irq.h>
 #include "serial.h"
 #include "plat.h"
-#include "server_virtqueue.h"
 
 #define ESCAPE_CHAR '@'
 #define MAX_CLIENTS 12
-#define CLIENT_OUTPUT_BUFFER_SIZE 4096
-
-/* TODO: have the MultiSharedData template generate a header with these */
-void getchar_emit(unsigned int id) WEAK;
-seL4_Word getchar_enumerate_badge(unsigned int id) WEAK;
-unsigned int getchar_num_badges() WEAK;
-void *getchar_buf(unsigned int id) WEAK;
-void *processed_batch_buf(unsigned int id) WEAK;
-void *raw_batch_buf(unsigned int id) WEAK;
-int getchar_largest_badge(void) WEAK;
-
-typedef struct getchar_buffer {
-    uint32_t head;
-    uint32_t tail;
-    char buf[4096 - 8];
-} client_buffer_t;
-
-compile_time_assert(getchar_buf_sized, sizeof(client_buffer_t) == sizeof(Buf));
+#define BUFSIZE 4088
```

Comment:
Can you add a note that explains why this is 4088?

Reply:
Or, as it has to be one small page, just use the PAGESIZE macro.

Reply:
Don't we have something like `PAGESIZE' here, so there is no need to refer to a magic values `0x1000`. It really helps if there is some explanation in the code what really drives the sizes in the end. And referring to constants is much better than plain values.